src/map.c

```diff
@@ -111,21 +111,15 @@ SEXP map2_impl(SEXP env, SEXP x_name_, SEXP y_name_, SEXP f_name_, SEXP type_) {
   check_vector(y_val, ".y");
 
   int nx = Rf_length(x_val), ny = Rf_length(y_val);
-  if (nx == 0 || ny == 0) {
-    SEXP out = PROTECT(Rf_allocVector(type, 0));
-    copy_names(x_val, out);
-    UNPROTECT(3);
-    return out;
-  }
-  if (nx != ny && !(nx == 1 || ny == 1)) {
+  if (nx != ny && nx != 1 && ny != 1) {
```

Comment:
Can we unclass data frames and use `vec_recycle_common()` instead?

Reply:
I thought about that too, but I worried (possibly needlessly) about the performance implications of now `rep`-ing scalars into large vectors instead. And the set of changes seemed small enough.

Reply:
I think it's fine merging as is. Just noting that in the future the rep'd scalars would become altrep repetitions so the problem would go away eventually.